src/main/java/com/BlogSite/TestBlogProject/services/PostServiceImpl.java

```diff
@@ -21,12 +21,11 @@ public class PostServiceImpl implements PostService {
     @Override
     public Result<List<Post>> getPostsByUsername(String username) {
         Result<List<Post>> result = new Result<>();
-        User user = userService.getUserByUsername(username).getData();
-        if (user == null) {
+        List<Post> data = postRepository.findByUserUsername(username);
+        if (data == null)
```

Comment:
also i recommend you to install https://plugins.jetbrains.com/plugin/7973-sonarlint this plugin.
Its the most famous static code analyzer. It would help you to maintain quality of code